src/types.ts

```diff
@@ -0,0 +1,14 @@
+export type Resource = {
```

Comment:
type makes more sense if we aren't using classes I think?

Reply:
Yes. `Interface` is strictly for OOP patterns IMO.